### PR TITLE
Bump k8s.io/client-go from 0.29.3 to 0.30.0

### PR DESCRIPTION
**Pull request by benashz**
Updated with:
```
go get k8s.io/client-go@v0.30.0
```

This update  automatically bumped the go version to 1.22.0 in go.mod:
```
$ go get k8s.io/client-go@v0.30.0
go: upgraded go 1.21 => 1.22.0
go: added toolchain go1.22.2
go: upgraded k8s.io/api v0.29.3 => v0.30.0
go: upgraded k8s.io/apimachinery v0.29.3 => v0.30.0
go: upgraded k8s.io/client-go v0.29.3 => v0.30.0
go: upgraded k8s.io/klog/v2 v2.110.1 => v2.120.1
go: upgraded k8s.io/kube-openapi v0.0.0-20231010175941-2dd684a91f00 => v0.0.0-20240228011516-70dd3763d340

```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -28,7 +30,7 @@ require (
 	google.golang.org/api v0.176.1
 	gopkg.in/yaml.v3 v3.0.1
 	k8s.io/api v0.29.3
-	k8s.io/apimachinery v0.29.3
+	k8s.io/apimachinery v0.30.0
 	k8s.io/client-go v0.29.3
```
Comment (tvoran) [inline go.mod:34]: The title says "Bump k8s.io/client-go" but we're not bumping client-go?

File: go.mod [hunk 2]
```diff
@@ -1,6 +1,8 @@
 module github.com/hashicorp/vault-secrets-operator
 
-go 1.21
+go 1.22.0
```
Comment (tvoran) [inline go.mod:3]: Odd that [apimachinery](https://github.com/kubernetes/apimachinery/blame/v0.30.0/go.mod) is forcing 1.22.0 here 🤷

**Review by tvoran — APPROVED** [review]
May want to update the other k8s packages too?

File: go.mod (hunk 1 above)
Comment (benashz) [reply go.mod]: Good catch - fixed in 4b8f88adbb259b81b4d6c40b886053254b48dec2

File: go.mod (hunk 2 above)
Comment (benashz) [reply go.mod:3]: Yeah, same for [client-go](https://github.com/kubernetes/client-go/blame/master/go.mod)

Comment (benashz) [reply go.mod:3]: Things still look good either way.

**Review by thyton — APPROVED** [review]

**Comment by benashz** [comment]
@tvoran - bumping controller-runtime to v0.18.0 was required for this change:

```
$ go get sigs.k8s.io/controller-runtime@v0.18.0
go: downloading sigs.k8s.io/controller-runtime v0.18.0
go: downloading github.com/evanphx/json-patch/v5 v5.9.0
go: downloading k8s.io/apiextensions-apiserver v0.30.0
go: downloading k8s.io/component-base v0.30.0
go: upgraded github.com/evanphx/json-patch/v5 v5.8.0 => v5.9.0
go: upgraded k8s.io/apiextensions-apiserver v0.29.2 => v0.30.0
go: upgraded k8s.io/component-base v0.29.2 => v0.30.0
go: upgraded sigs.k8s.io/controller-runtime v0.17.3 => v0.18.0
```

This upgrade brought in some breaking changes that were handled in 2f0d610554fc18bcedfca62e178e17b2ec32091c

**Pull request merged**